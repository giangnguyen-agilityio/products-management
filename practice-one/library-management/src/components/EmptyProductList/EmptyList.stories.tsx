import { type StoryFn, type Meta } from "@storybook/react";
import EmptyProductList from "./index";

export default {
  title: "Example/EmptyProductList",
  component: EmptyProductList,
} as Meta<typeof EmptyProductList>;

const Template: StoryFn<typeof EmptyProductList> = (args) => <EmptyProductList {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  errorMessage: 'We couldn\'t find any books at the moment',
};