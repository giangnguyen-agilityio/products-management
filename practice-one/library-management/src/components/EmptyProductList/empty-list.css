.empty-product-list {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.error-list-title {
  display: flex;
  text-align: center;
  width: 100%;
  justify-content: center;
  font-family: var(--secondary-font);
  font-size: 30px;
  color: var(--primary-color);
  padding-bottom: 25px;
}
