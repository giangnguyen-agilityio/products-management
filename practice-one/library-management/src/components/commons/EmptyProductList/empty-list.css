.empty-product-list {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.error-list-title {
  display: flex;
  text-align: center;
  width: 100%;
  justify-content: center;
  font-family: var(--secondary-font);
  font-size: 30px;
  color: var(--primary-color);
  padding-bottom: 25px;
}

/* Media Query for Laptop Devices */
@media only screen and (max-width: 1024px) {
  .error-list-image {
    height: 100%;
  }

  .error-list-title {
    font-size: 28px;
  }
}

/* Media Query for Tablets and Smaller Screens */
@media only screen and (max-width: 768px) {
  .error-list-title {
    font-size: 24px;
  }
}

/* Media Query for Mobile Devices */
@media only screen and (max-width: 480px) {
  .error-list-image img {
    width: 350px;
  }
  .error-list-title {
    padding-top: 25px;
    font-size: 20px;
  }
}
