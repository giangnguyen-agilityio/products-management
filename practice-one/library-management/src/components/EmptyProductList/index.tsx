import React from 'react'

// Importing the image URL as the variable
import errorImage from '../../assets/images/error-image.png'

// Importing the Typography component
import Typography from '../Typography/index'

// Importing the CSS file for styling
import './empty-list.css'

// Define the props for the EmptyProductList component
interface EmptyProductListProps {
  errorMessage?: string
}

const EmptyProductList: React.FC<EmptyProductListProps> = (props) => {
  const { errorMessage } = props
  return (
    <div className="empty-product-list">
      <div className="error-list-image">
        <img src={errorImage} alt="This is the error image" />
      </div>
      <Typography variant="p" className="error-list-title">
        {errorMessage}
      </Typography>
    </div>
  )
}

export default EmptyProductList
