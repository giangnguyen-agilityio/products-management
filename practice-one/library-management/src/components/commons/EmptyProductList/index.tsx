import React from 'react'
import errorImage from '@assets/images/error-image.png'
import Typography from '@components/commons/Typography'
import './empty-list.css'

interface EmptyProductListProps {
  errorMessage?: string
}

const EmptyProductList: React.FC<EmptyProductListProps> = props => {
  const {errorMessage} = props
  return (
    <div className="empty-product-list">
      <div className="error-list-image">
        <img src={errorImage} alt="This is the error image" />
      </div>
      <Typography variant="p" className="error-list-title">
        {errorMessage}
      </Typography>
    </div>
  )
}

export default EmptyProductList
