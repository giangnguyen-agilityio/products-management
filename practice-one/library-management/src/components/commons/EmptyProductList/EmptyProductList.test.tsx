import React from 'react'
import {render} from '@testing-library/react'
import EmptyProductList from '.'

jest.mock('@assets/images/error-image.png', () => ({
  default: 'mocked-error-image',
}))

describe('EmptyProductList component', () => {
  it('should renders correctly with error message prop and the error image', () => {
    const errorMessage = 'This is an error message'
    const {container} = render(<EmptyProductList errorMessage={errorMessage} />)

    // Assert that the component renders the error message
    expect(container).toHaveTextContent(errorMessage)

    // Assert that the component renders the mocked image correctly
    const imgElement = container.querySelector('img')
    expect(imgElement).toBeInTheDocument()
    expect(imgElement?.getAttribute('src')).toBe('mocked-error-image')

    // Snapshot testing
    expect(container).toMatchSnapshot()
  })
})
