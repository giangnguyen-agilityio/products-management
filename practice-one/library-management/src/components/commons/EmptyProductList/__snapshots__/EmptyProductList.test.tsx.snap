// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyProductList component should renders correctly with error message prop and the error image 1`] = `
<div>
  <div
    class="empty-product-list"
  >
    <div
      class="error-list-image"
    >
      <img
        alt="This is the error image"
        src="mocked-error-image"
      />
    </div>
    <p
      class="text-p text-default error-list-title"
    >
      This is an error message
    </p>
  </div>
</div>
`;
